// @sency/react-native-smkit-ui/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AssessmentTypes = void 0;
exports.configure = configure;
exports.startAssessment = startAssessment;
exports.startCustomWorkout = startCustomWorkout;
exports.startWorkoutProgram = startWorkoutProgram;
var _reactNative = require("react-native");
const LINKING_ERROR = `The package 'react-native-smkit-ui' doesn't seem to be linked. Make sure: \n\n` + _reactNative.Platform.select({
  ios: "- You have run 'pod install'\n",
  default: ''
}) + '- You rebuilt the app after installing the package\n' + '- You are not using Expo Go\n';
const SMKitUIManager = _reactNative.NativeModules.SMKitUIManager ? _reactNative.NativeModules.SMKitUIManager : new Proxy({}, {
  get() {
    throw new Error(LINKING_ERROR);
  }
});
function configure(key) {
  return SMKitUIManager.configure(key);
}
function startAssessment(type, showSummary) {
  return SMKitUIManager.startAssessment(type, showSummary);
}
function startCustomWorkout(workout) {
  return SMKitUIManager.startCustomWorkout(workout);
}
function startWorkoutProgram(workoutConfig) {
  return SMKitUIManager.startWorkoutProgram(workoutConfig);
}
let AssessmentTypes = exports.AssessmentTypes = /*#__PURE__*/function (AssessmentTypes) {
  AssessmentTypes["Fitness"] = "fitness";
  AssessmentTypes["Custom"] = "custom";
  return AssessmentTypes;
}({});
