// @sency/react-native-smkit-ui/src/SMWorkout.tsx
export enum AssessmentTypes{
  Fitness = "fitness"
}

export enum UIElement{
  RepsCounter = "repsCounter",
  Timer = "timer",
  GaugeOfMotion = "gaugeOfMotion"
}

export enum BodyZone{
  UpperBody = "UpperBody",
  LowerBody = "LowerBody",
  FullBody = "FullBody"
}

export enum WorkoutDifficulty{
  LowDifficulty = "LowDifficulty",
  MidDifficulty = "MidDifficulty",
  HighDifficulty = "HighDifficulty"
}

export enum WorkoutDuration{
  Short = "Short",
  Long = "Long"
}

export class SMWorkout{
 id: string | null;
 name: string | null;
 workoutIntro: string | null;
 soundtrack: string | null;
 exercises: SMExercise[];
 getInFrame: string | null;
 bodycalFinished: string | null;
 workoutClosure: string | null;

 constructor(id: string | null, name: string | null, workoutIntro: string | null, soundtrack: string | null, exercises: SMExercise[], getInFrame: string | null, bodycalFinished: string | null, workoutClosure: string | null) {
        this.id = id || null;
        this.name = name || null;
        this.workoutIntro = workoutIntro || null;
        this.soundtrack = soundtrack || null;
        this.exercises = exercises;
        this.getInFrame = getInFrame || null;
        this.bodycalFinished = bodycalFinished || null;
        this.workoutClosure = workoutClosure || null;
    }
}

export class SMExercise{
  name:string | null;
  totalSeconds: number | null;
  introSeconds: number | null;
  videoInstruction: string | null;
  exerciseIntro: string | null;
  uiElements: UIElement[] | null;
  detector: string;
  repBased: boolean | null;
  exerciseClosure: string | null;
  targetReps: number | null;
  targetTime: number | null;
  scoreFactor: number | null;

  constructor(name: string | null, totalSeconds: number | null, introSeconds: number | null, videoInstruction: string | null, exerciseIntro: string | null, uiElements: UIElement[] | null, detector: string, repBased: boolean | null, exerciseClosure: string | null, targetReps: number | null, targetTime: number | null, scoreFactor: number | null) {
    this.name = name || null
    this.totalSeconds = totalSeconds || null
    this.introSeconds = introSeconds || null
    this.videoInstruction = videoInstruction || null
    this.exerciseIntro = exerciseIntro || null
    this.uiElements = uiElements || null
    this.detector = detector
    this.repBased = repBased || null
    this.targetReps = targetReps || null
    this.targetTime = targetTime || null
    this.scoreFactor = scoreFactor || null
    this.exerciseClosure = exerciseClosure || null;
  }
}

export class WorkoutConfig{
  week: number;
  bodyZone: BodyZone;
  difficultyLevel: WorkoutDifficulty;
  workoutDuration: WorkoutDuration;
  programID: string;

  constructor(week: number, bodyZone: BodyZone, difficultyLevel: WorkoutDifficulty, workoutDuration: WorkoutDuration, programID: string){
    this.week = week;
    this.bodyZone = bodyZone;
    this.difficultyLevel = difficultyLevel;
    this.workoutDuration = workoutDuration;
    this.programID = programID;
  }
}
