// @sency/react-native-smkit-ui/lib/module/index.js
import { NativeModules, Platform } from 'react-native';
const LINKING_ERROR = `The package 'react-native-smkit-ui' doesn't seem to be linked. Make sure: \n\n` + Platform.select({
  ios: "- You have run 'pod install'\n",
  default: ''
}) + '- You rebuilt the app after installing the package\n' + '- You are not using Expo Go\n';
const SMKitUIManager = NativeModules.SMKitUIManager ? NativeModules.SMKitUIManager : new Proxy({}, {
  get() {
    throw new Error(LINKING_ERROR);
  }
});
export function configure(key) {
  return SMKitUIManager.configure(key);
}
export function startAssessment(type, showSummary) {
  return SMKitUIManager.startAssessment(type, showSummary);
}
export function startCustomWorkout(workout) {
  return SMKitUIManager.startCustomWorkout(workout);
}
export function startWorkoutProgram(workoutConfig) {
  return SMKitUIManager.startWorkoutProgram(workoutConfig);
}
export let AssessmentTypes = /*#__PURE__*/function (AssessmentTypes) {
  AssessmentTypes["Fitness"] = "fitness";
  AssessmentTypes["Custom"] = "custom";
  return AssessmentTypes;
}({});
