// @sency/react-native-smkit-ui/lib/module/SMWorkout.js
export let AssessmentTypes = /*#__PURE__*/function (AssessmentTypes) {
  AssessmentTypes["Fitness"] = "fitness";
  return AssessmentTypes;
}({});
export let UIElement = /*#__PURE__*/function (UIElement) {
  UIElement["RepsCounter"] = "repsCounter";
  UIElement["Timer"] = "timer";
  UIElement["GaugeOfMotion"] = "gaugeOfMotion";
  return UIElement;
}({});
export let BodyZone = /*#__PURE__*/function (BodyZone) {
  BodyZone["UpperBody"] = "UpperBody";
  BodyZone["LowerBody"] = "LowerBody";
  BodyZone["FullBody"] = "FullBody";
  return BodyZone;
}({});
export let WorkoutDifficulty = /*#__PURE__*/function (WorkoutDifficulty) {
  WorkoutDifficulty["LowDifficulty"] = "LowDifficulty";
  WorkoutDifficulty["MidDifficulty"] = "MidDifficulty";
  WorkoutDifficulty["HighDifficulty"] = "HighDifficulty";
  return WorkoutDifficulty;
}({});
export let WorkoutDuration = /*#__PURE__*/function (WorkoutDuration) {
  WorkoutDuration["Short"] = "Short";
  WorkoutDuration["Long"] = "Long";
  return WorkoutDuration;
}({});
export class SMWorkout {
  constructor(id, name, workoutIntro, soundtrack, exercises, getInFrame, bodycalFinished, workoutClosure) {
    this.id = id || null;
    this.name = name || null;
    this.workoutIntro = workoutIntro || null;
    this.soundtrack = soundtrack || null;
    this.exercises = exercises;
    this.getInFrame = getInFrame || null;
    this.bodycalFinished = bodycalFinished || null;
    this.workoutClosure = workoutClosure || null;
  }
}
export class SMExercise {
  constructor(name, totalSeconds, introSeconds, videoInstruction, exerciseIntro, uiElements, detector, repBased, exerciseClosure, targetReps, targetTime, scoreFactor) {
    this.name = name || null;
    this.totalSeconds = totalSeconds || null;
    this.introSeconds = introSeconds || null;
    this.videoInstruction = videoInstruction || null;
    this.exerciseIntro = exerciseIntro || null;
    this.uiElements = uiElements || null;
    this.detector = detector;
    this.repBased = repBased || null;
    this.targetReps = targetReps || null;
    this.targetTime = targetTime || null;
    this.scoreFactor = scoreFactor || null;
    this.exerciseClosure = exerciseClosure || null;
  }
}
export class WorkoutConfig {
  constructor(week, bodyZone, difficultyLevel, workoutDuration, programID) {
    this.week = week;
    this.bodyZone = bodyZone;
    this.difficultyLevel = difficultyLevel;
    this.workoutDuration = workoutDuration;
    this.programID = programID;
  }
}
